frontend/catalyst/compiler.py

```diff
@@ -179,6 +178,7 @@ def run_writing_command(command: List[str], compile_options: Optional[CompileOpt
 MLIR_TO_LLVM_PASS = (
     "MLIRToLLVMDialect",
     [
+        "expand-realloc",
```

Comment:
Small comment why this pass is necessary will be nice, since it's hard to see why pipelines are the way they are.

Reply:
https://reviews.llvm.org/D159430 This was part of the memref lowering and it was extracted to make lowering simpler

Reply:
Right I meant adding it as a comment into the file